bingo/symbolic_regression/agraph/agraph.py

```diff
@@ -104,22 +114,42 @@ class AGraph(Equation, continuous_local_opt.ChromosomeInterface):
         to be made.
     constants : tuple of numeric
         numeric constants that are used in the equation
-
     """
-    def __init__(self, use_simplification=False):
+    def __init__(self, use_simplification=False, sympy_representation=None):
         super().__init__()
-        self._command_array = np.empty([0, 3], dtype=int)
-
-        self._simplified_command_array = np.empty([0, 3], dtype=int)
-        self._simplified_constants = []
 
-        self._needs_opt = False
-        self._modified = False
         self._use_simplification = use_simplification
 
         if use_simplification and not USING_PYTHON_SIMPLIFICATION:
             force_use_of_python_simplification()
 
+        if sympy_representation is None:
+            self._command_array = np.empty([0, 3], dtype=int)
+
+            self._simplified_command_array = np.empty([0, 3], dtype=int)
+            self._simplified_constants = []
+
+            self._needs_opt = False
+            self._modified = False
+        elif isinstance(sympy_representation, (Expr, str)):
+            if isinstance(sympy_representation, Expr):
+                self._sympy_expr = sympy_representation
+                self._sympy_str = str(sympy_representation)
+            else:  # is str instance
+                self._sympy_expr = parse_expr(sympy_representation)
+                self._sympy_str = str(self._sympy_expr)
+                # not using sympy_representation directly for self._sympy_str
+                # because it might not be in simplified form
+            command_array, constants = \
+                sympy_string_to_command_array_and_constants(self._sympy_str)
+            self.set_local_optimization_params(constants)
+            self.command_array = command_array
+            # TODO decide on setting _needs_opt or not
```

Comment:
A couple of considerations for setting up the AGraph from a Sympy string. If we get a string like "1.0 + 2.0", we get an AGraph that adds two constants together and sets those constants at 1.0 and 2.0 respectively. Should _needs_opt be true in this case or should it be false like normal and be left for the user to decide?

Reply:
agreed, made needs_opt = True the default in the new changes